CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+
+Once you have selected an issue you'd like to work on, make sure it's not already assigned to someone else. Then, remember to assign it to yourself, by clicking on "Assign to me", to prevent someone else from also working on the same issue.
+
+![jira](assets/images/jira.png)
+
+It is recommend that you use a separate branch for every Pull Request to submit. To keep things straightforward and memorable, you should try to name the branch the JIRA issue number. Doing so also allows your to prevent conflicts in the future when working on other issues. For example, for ELY-2127, you should also name the branch ELY-2127.
+
+## Setting up your Developer Environment
+You will need:
+
+* JDK 11
+* Git
+* Maven 3.3.9 or later
+* An [IDE](https://en.wikipedia.org/wiki/Comparison_of_integrated_development_environments#Java)
+(e.g., [IntelliJ IDEA](https://www.jetbrains.com/idea/download/), [Eclipse](https://www.eclipse.org/downloads/), etc.)
+
+First `cd` to the directory where you cloned the project (eg: `cd wildfly-elytron`)
+
+Add a remote ref to upstream, for pulling future updates.
+For example:
+
+```
+git remote add upstream https://github.com/wildfly-security/wildfly-elytron
+```
+To build `wildfly-elytron` run:
+```bash
+mvn clean install
+```
+
+To skip the tests, use:
+
+```bash
+mvn clean install -DskipTests=true
+```
+
+To run only a specific test, use:
+
+```bash
+mvn clean install -Dtest=TestClassName
+```
+For more information, including details on how WildFly Elytron is integrated in WildFly Core and WildFly, check out our [developer guide](https://wildfly-security.github.io/wildfly-elytron/getting-started-for-developers/).
+
+## Coding Guidlines
+
+When submitting a PR, please keep the following guidelines in mind:
+
+1. In general, it's good practice to squash all of your commits into a single commit. For larger changes, it's ok to have multiple meaningful commits. If you need help with squashing your commits, feel free to ask us how to do this on your pull request. We're more than happy to help!
+
+2. Please include the JIRA issue you worked on in the title of your pull request and in your commit message. For example, for [ELY-2127](https://issues.redhat.com/browse/ELY-2127), the PR title and commit message should be `[ELY-2127] Add client side TRACE logging after successful authentication`.
+
+3. Please include the link to the JIRA issue you worked on in the description of the pull request. For example, if your PR adds a fix for [ELY-2127](https://issues.redhat.com/browse/ELY-2127), the PR description should contain a link to https://issues.redhat.com/browse/ELY-2127.
+
+For an example of a properly formatted PR, take a look at https://github.com/wildfly-security/wildfly-elytron/pull/1532
```

Comment:
Maybe we could add a section at the end that says:

## Community

For more information on how to get involved with WildFly Elytron, check out our [community](https://wildfly-security.github.io/wildfly-elytron/community/) page.

Reply:
Added